crates/meroctl/src/cli/context/create.rs

```diff
@@ -0,0 +1,190 @@
+use calimero_primitives::application::ApplicationId;
+use calimero_primitives::identity::Context;
+use calimero_server_primitives::admin::ApplicationListResult;
+use camino::Utf8PathBuf;
+use clap::{ArgGroup, Parser};
+use reqwest::{Client, Url};
+use semver::Version;
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use tracing::info;
+
+use crate::cli::context::common::get_ip;
+use crate::cli::RootArgs;
+use crate::config_file::ConfigFile;
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateContextRequest {
+    application_id: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CreateContextResponse {
+    data: Context,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct InstallApplicationRequest {
+    pub application: ApplicationId,
+    pub version: Version,
+    pub path: Option<Utf8PathBuf>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ListApplicationsResponse {
+    data: ApplicationListResult,
+}
+
+#[derive(Debug, Parser)]
+#[clap(group(
+    ArgGroup::new("dev_args")
+        .multiple(true)
+        .requires_all(&["dev", "path", "version"])
+))]
+pub struct CreateCommand {
+    /// The application ID to attach to the context
+    #[clap(
+        long,
+        short = 'a',
+        default_value = "",
+        exclusive = true,
+        value_name = "APP_ID"
+    )]
+    application_id: String,
+
+    /// Enable dev mode
+    #[clap(long, group = "dev_args")]
+    dev: bool,
+
+    /// Path to use in dev mode
+    #[clap(
+        short,
+        long,
+        group = "dev_args",
+        default_value = "",
+        value_name = "PATH"
+    )]
+    path: Utf8PathBuf,
+
+    /// Version of the application
+    #[clap(
+        short,
+        long,
+        group = "dev_args",
+        default_value = "0.0.0",
+        value_name = "VERSION"
+    )]
+    version: Version,
+}
+
+impl CreateCommand {
+    pub async fn run(self, root_args: RootArgs) -> eyre::Result<()> {
+        let path = root_args.home.join(&root_args.node_name);
+
+        if !ConfigFile::exists(&path) {
+            eyre::bail!("Config file does not exist")
+        };
+
+        let Ok(config) = ConfigFile::load(&path) else {
+            eyre::bail!("Failed to load config file")
+        };
+
+        let Some(multiaddr) = config.network.server.listen.first() else {
+            eyre::bail!("No address.")
+        };
+
+        let base_url = get_ip(multiaddr)?;
+
+        match self.dev {
+            true => Ok(link_local_app(base_url, self.path, self.version).await?),
+            false => Ok(create_context(base_url, self.application_id).await?),
+        }
```

Comment:
an `if` statement here reads better, no?

```suggestion
        if self.dev {
            return link_local_app(base_url, self.path, self.version).await
        }

        create_context(base_url, self.application_id).await
```